src/main/java/HttpServer.java

```diff
@@ -11,7 +11,8 @@ public static void main(String[] args) throws IOException {
     SocketIO socket = new SocketIO(arguments.getPortNumber());
     HttpReaderWriter httpReaderWriter = new HttpReaderWriter(socket, builder);
     FileIO fileIO = new FileIO(arguments.getDirectory());
-    HttpHandlerSelector handler = new HttpHandlerSelector(fileIO);
+    HandleGetRequest getRequest = new HandleGetRequest(fileIO);
+    HttpHandlerSelector handler = new HttpHandlerSelector(getRequest);
```

Comment:
Consider what might happen in the future here. If your `HttpHandlerSelector` needs to be constructed with one of each type of request handler, that means that on every request you're constructing handlers for all types of requests, only to discard all but one of them. Seems a little wasteful, no? I think what you're looking for is something like a Router or a HandlerRegistry...